src/Repository/AvailableFile.hpp

```diff
@@ -54,6 +54,12 @@ struct AvailableFile {
 
   BrokenDate update_date;
 
+  /**
+  * The SHA256 hash of the contents of this file.
+  */
+  std::array<std::byte, 32> sha256_hash;
+  bool has_hash;
```

Comment:
This `bool` takes 8 bytes. Can we do better than that? Maybe consider the hash to be unavailable when it's all zeroes?

Reply:
I thought about that too, but I wanted to be on the safe side.
Given how unlikely it is for a file to have a hash of all zeroes it should work though.

Reply:
Went with testing for zeroes now, so I added a `HasHash` function.